code/modules/mob/living/carbon/human/human.dm

```diff
@@ -1465,17 +1465,24 @@
 	nutrition_loss = HUNGER_FACTOR * species.nutrition_loss_factor
 	hydration_loss = THIRST_FACTOR * species.hydration_loss_factor
 
+	fill_out_culture_data_random()
+
 	if(change_hair)
 		species.set_default_hair(src)
 
-	if(species.default_accent)
-		accent = species.default_accent
-
 	if(species)
 		return 1
 	else
 		return 0
 
+
+/mob/living/carbon/human/proc/fill_out_culture_data_random()
+	culture = decls_repository.get_decl(pick(species.possible_cultures))
+	origin = decls_repository.get_decl(pick(culture.possible_origins))
+	accent = pick(origin.possible_accents)
+	citizenship = pick(origin.possible_citizenships)
+	religion = pick(origin.possible_religions)
```

Comment:
Don't think this should ever be random as discussed elsewhere.

Reply:
This is different - this is for mobs that are spawned without prefs, e.g in game. Does it matter then?

Reply:
I think it does. Mobs like that do not allow much customization, and _usually_ no one cares what culture/origin such a ghost-role character has -- So default (choose the least restrictive culture) is perfect? Otherwise you will sometimes have e.g. Elyran Orion Couriers that don't fit the rules for name / colour prefs.

Reply:
The appearance menu you get when spawning as a ghost role or antag already covers changing origins/backgrounds/accents, though.